### DISCUSSION TITLE
Make exported tids much more useful

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5456] [Ideas]
As an end user I'd like for the exported tiddler to fullfil all of these:

0. Self-contained and viewable in regular browser
1. Same appearance as when viewed in TW
2. Simple to edit the tiddler content in a regular external editor
3. Re-importable into a single-file TW and it becomes/works like a regular tiddler
4. As "integrateable" as possible with a single-file TW, without it actually being imported. If I estimate complexity I guess this would mean 1) viewable 2) editable 3) interactionable with other tids

Because of "1. Same appearance as normal" I would assume it then **has** to be in HTML/CSS... which would mean the question really is; Is the other stuff theoretically achievable with a HTML tiddler?

**Comment by Jermolene** [comment]
Hi @twMat I think that implies that the file would contain two representations of the tiddler: one in wikitext for editing and subsequent importing, and one in static HTML for viewing. If so, the challenge is what happens when those two representations get out of sync due to manual editing. One could imagine the HTML version using JavaScript to verify a hash to detect whether the wikitext content has changed, and to display a warning message. But I'm still not sure how useful it would be...

**Reply by twMat** [reply]
Thanks for feedback!

Your comment seems to address bullet "4. integrateable ....", specifically the most complex steps therein.

But how do you see the bullets 1-3?
Perhaps I was ambiguous: "Re-importing" really would only need to mean to make a *copy* in the wiki, i.e either that the import mechanism copies and converts the html into wikitext or, more easily but less good, it just extracts the single-tiddlers HTML content (and possibly also the CSS + wraps with style tags.) Thus "re-importing" and editing really leaves the single-tiddler-file untouched and I don't mean any syncing of data.

The point is this: It is really cool to be able to export and send off tiddlers to non-TW users. For example, in my work as a science teacher I add single tiddlers with notes and stuff on subjects for my colleagues, into our shared Drive folders. It would be neat if I could easily tweak old such tiddlers though at later date. It would be fine to re-import it (i.e copy), edit, and export again and delete that original shared single tiddler.

Currently, if I anyone wants to edit an old single HTML tiddler, he must either 
* access the original wiki and locate it there 
* dig around in the html which is messy mostly because of the CSS overload
* remake the tiddler from scratch in a new wiki.

The other single tiddler formats have contrasting weaknesses and strengths.

**Comment by Parlazo** [comment]
> * Self-contained and viewable in regular browser
> * Same appearance as when viewed in TW
> * Simple to edit the tiddler content in a regular external editor
> * Re-importable into a single-file TW and it becomes/works like a regular tiddler
> * As "integrateable" as possible with a single-file TW, without it actually being imported. If I estimate complexity I guess this would mean 1) viewable 2) editable 3) interactionable with other tids

I like this idea too
may the whole Wiki should get exported but only with the single selected or mass selected tiddler and optionaly with all plugins and styles and every thing or just as an empty TiddlerWiki with the selected Tiddlers

in such a way it would look the same and it would be editable in the TiddlerWiki or with copy/paste in an external editor.

**Comment by twMat** [comment]
Just got an idea (that I have no idea how to implement... plus it's late so maybe it's just a stupid idea...)

Could there be a **"wrap" that takes raw tiddler content and presents it as a tiddler**? It might seem this would break the first point;

0. Self-contained and viewable in regular browser

...but maybe the single exported tiddler **file could really contain two sections**; the wrap (generated in export from some static template) which somehow presents the data that appears in another section, to give the content the...

1. Same appearance as when viewed in TW

...and this would make it very easy to both

2. Simple to edit the tiddler content in a regular external editor
3. Re-importable into a single-file TW and it becomes/works like a regular tiddler

Depending on how it is implemented (and if at all possible) then maybe even some of the last point...

4. As "integrateable" as possible with a single-file TW, without it actually being imported. If I estimate complexity I guess this would mean 1) viewable 2) editable 3) interactionable with other tids
